app/views/apipie_dsl/apipie_dsls/help.html.erb

```diff
@@ -1,58 +1,340 @@
+
+<table class="table table-condensed">
+  <thead>
+  <tr>
+    <th><%= _('Template type') %></th>
+    <th><%= _('Available input types') %></th>
+  </tr>
+  </thead>
+  <tbody>
+  <tr>
+    <td>Provisioning template</td>
+    <td>Fact value, Variable, Puppet parameter</td>
+  </tr>
+  <tr>
+    <td>Partition table</td>
+    <td>Fact value, Variable, Puppet parameter</td>
+  </tr>
+  <tr>
+    <td>Report template</td>
+    <td>User input</td>
+  </tr>
+  <tr>
+    <td>Job template</td>
+    <td>User input, Fact value, Variable, Puppet parameter</td>
+  </tr>
+  </tbody>
+</table>
+
+<h2 id="snippets"><%= _('Snippets') %></h2>
+
+<p><%= _('Snippets are special kind of templates that can be easily reused in multiple other templates. Imagine there is a bootstrap procedure to be run on every provisioned host. At the same time, there are multiple kickstart templates. To avoid duplication a snippet can be used. Assuming there is a snippet created with a name Bootstrap, following example will inject it in a template') %></p>
+
+<%= apipie_dsl_example(apipie_erb_wrap("snippet('Bootstrap')")) %>
+
+<h2 id="safe_mode"><%= _('Safe mode') %></h2>
+
+<p><%= _('Safe Mode is how we refer to the safety mechanism we have to prevent doing harmful things. It whitelists Ruby methods, that can be used in the template. When a safe mode protected method is called accidentally, rendering fails with the error that says "Safemode doesn\'t allow to access ...".') %></p>
+
+<p><%= _('When writing a template, prefer to use macros and variables, that are already available with Safe Mode enabled. If you are looking for some information that you are not able to retrieve via allowed methods or macros listed in this documentation, you can consider disabling the Safe Mode. Doing that, you will be able to access all internal application variables and methods directly, when the template is rendered.') %></p>
+
+<p><%= _('Be aware, this is dangerous and it can cause performance and security issues. Disabling safe mode allows any Ruby code to be executed on the server. Any application user that can edit the template can modify the template, so that when it is rendered, it spawns a shell process and run arbitrary command in it. Such commands would be ran under the "foreman" user and would still be restricted by SELinux policy. Disabling Safe Mode is only recommended in trusted environment.') %></p>
+
+<h3 id="disabling_safe_mode"><%= _('Disabling Safe mode') %></h3>
+
+<p><%= (_('In order to disable Safe mode, navigate to Administer -> %{settings_link} -> Provisioning tab and search for a setting called "Safemode rendering". Value "No" means, rendering is unprotected and potentially dangerous. This installation has currently safe mode: <b>%{status}</b>.') % {:status => (Setting[:safemode_render] ? _('enabled') : _('disabled')), :settings_link => link_to(_('Settings'), settings_path)}).html_safe %></p>
```

Comment:
`settings_path` is also unavailable within apipie-dsl renderer. It creates its own instead of using Rails' one, which has more metadata such as helper methods and path methods.